Upcoming events

albachteng opened the pull request:
First pull request, so thanks in advance for taking a look and for your patience! 

Updated styles in main.scss to: 

- extend margin below header and link to 48px
- update font styles for online event and attend-button button (including font-awesome icon)
- update link and hover styles for attend-button

jrapala commented on scss/main.scss:
```diff
@@ -295,15 +298,28 @@ h1,h2,h3 {
 .rsvp-stuff {
   margin-top: 1rem;
+  p {
+    font-family: 'Work sans', sans-serif;
```
Let's use the `$font-family-alt` variable instead. In `.attend-button` as well.

jrapala commented on scss/main.scss:
```diff
@@ -295,15 +298,28 @@ h1,h2,h3 {
+  p {
+    font-family: 'Work sans', sans-serif;
+    font-weight: 500;
+    color: #333;
```
Should use the `$darker-grey` variable instead. In `.attend-button` as well.

jrapala commented on scss/main.scss:
```diff
@@ -295,15 +298,28 @@ h1,h2,h3 {
   border-bottom: 1rem solid transparent;
   padding-left: 5%;
+  a:link, a:hover {
+    color: #0D77AC;
```
Let's use `$cfa-blue-dark` instead

jrapala reviewed (changes requested):
Left some comments about using variables instead of hard-coded values. It'll make it easier if we need to make some changes to our theme in the future.

Couple other things:

1. Can you also double-check the style of the attend button? It doesn't quite match the design:

**PR:**

![Screen Shot 2021-02-06 at 2 51 43 PM](https://user-images.githubusercontent.com/9326234/107129457-1eb73c80-688b-11eb-80f3-21797fc0e270.png)

![Screen Shot 2021-02-06 at 2 53 36 PM](https://user-images.githubusercontent.com/9326234/107129465-24ad1d80-688b-11eb-9cdf-2a7a033ea7dc.png)

**Design:**

![Screen Shot 2021-02-06 at 2 48 25 PM](https://user-images.githubusercontent.com/9326234/107129366-9fc20400-688a-11eb-88ae-88b8ceb9d601.png)

![Screen Shot 2021-02-06 at 2 48 00 PM](https://user-images.githubusercontent.com/9326234/107129473-3098df80-688b-11eb-912a-6b572629e173.png)

2. Can you check with Joseph to see how he wants to handle spacing between paragraphs in this section? There's currently a lot of spacing between these paragraphs:

![Screen Shot 2021-02-06 at 2 51 43 PM](https://user-images.githubusercontent.com/9326234/107129499-54f4bc00-688b-11eb-90d0-2f15312a4c56.png)

albachteng commented:
Hi Juliette! Thank you so much for reviewing for me! 

I updated the hard-coded values to variables and adjusted the border-radius on the button, but I'm going to need to spend some more time poking around to make the button respond to media-queries the way the design calls for. I will check with Joseph about the paragraph spacing in the event section.

albachteng commented:
Updated spacing between paragraphs within Upcoming Events section to 1rem per Joseph, corrected border on Attend button and icon. Updated colors and fonts with cfa- variables Only remaining issue is resizing the Attend button for smaller screens, which I'm drawing a blank on.

jrapala commented on scss/main.scss:
```diff
@@ -287,23 +287,48 @@ h1,h2,h3 {
+    font-family: $font-family-alt;
+    font-weight: 500;
+    color: $darker-grey;
+    font-size: 14px; 
```
Instead of hardcoded pixels, let's use REM instead, in case the user uses a larger default font.
14px equates to 0.875rem
Please adjust in `attend-button` as well.

Thanks!

albachteng commented:
Awesome, thanks again for your feedback and help!

jrapala commented on scss/main.scss:
```diff
@@ -287,23 +287,55 @@ h1,h2,h3 {
+  font-size: .875rem;
   border-bottom: 1rem solid transparent;
   padding-left: 5%;
+  @media (max-width: 361px) {
```
Can you please double-check this media query? The breakpoint should be a lot larger than 361px.

albachteng replied:
Absolutely! Double-checking now.

albachteng replied:
You were right, thank you! Joseph said 768px for now.

jrapala commented on scss/main.scss:
```diff
@@ -287,23 +287,55 @@ h1,h2,h3 {
 
 .attend-button {
+  font-family: $font-family-alt;
+  font-weight: 500;
```
Seems like this should be 600 instead

<img width="615" alt="Screen Shot 2021-02-10 at 9 49 51 PM" src="https://user-images.githubusercontent.com/9326234/107600693-18f28b80-6bea-11eb-8edd-bfb9651fc2cb.png">

albachteng replied:
Thank you! I'm just waiting on joseph for the font weight, the trello and figma appear to be in disagreement.

jrapala commented on scss/main.scss:
```diff
@@ -287,23 +287,55 @@ h1,h2,h3 {
+  color: $darker-grey; 
+  font-size: .875rem;
   border-bottom: 1rem solid transparent;
   padding-left: 5%;
```
This padding is making the button slightly off-center in the mobile view:

<img width="547" alt="Screen Shot 2021-02-10 at 9 29 05 PM" src="https://user-images.githubusercontent.com/9326234/107600846-8e5e5c00-6bea-11eb-94c7-29c752f2b566.png">

jrapala reviewed (changes requested):
Getting closer! Add a few more comments. We may need to trash your package-lock.json file. I brought up a question to the group in Slack.

albachteng commented:
Joseph asked me to change the attend button stylings to match donate button. I also adjusted the underline hover styles because they were extending beyond the text and it looked messy, but if anyone doesn't like that I can always revert it.

jrapala commented on scss/main.scss:
```diff
@@ -290,23 +290,60 @@ h1,h2,h3 {
+  }
+  
+  a:link, a:hover {
+    color: $white;
```
I think the anchor tag will always be white, correct?
We're currently not changing the :visited style, so the text disappears after you visit the link:

<img width="638" alt="Screen Shot 2021-02-18 at 8 33 49 AM" src="https://user-images.githubusercontent.com/9326234/108372796-c9512880-71c4-11eb-9d43-cc70ae02162b.png">

albachteng replied:
I am finding some strange behavior. It seems that something is overriding the color property of anchor tags throughout the file and turning them red, which has forced a lot of repetitive color rules like the one you quote above. When I remove the code you quote above or try to otherwise root out the repetition in the file, text starts turning red all over the place. I hate to leave the styles so repetitive, but the simplest solution I've found so far is just adding a:visited to the above rule, preventing the text from turning red. Any thoughts on what could be causing this?

daleinen7 replied:
I think what it should be is:
```suggestion
    a, a:link, a:hover {
      color: $white;
    }
```
Actually I don't know what a:link specifically is doing ... so maybe its redundant? But setting generic a tag seems to fix the behavior you're seeing on my end.

jrapala replied:
@albachteng You'll find the global anchor style in the `_reboot.scss` file. The color is currently set to `$cfa-red`. So, we'll need to override that, since this anchor seems to be a special case.

The `:link` pseudo-class targets any tag with an `href` attribute, that hasn't been visited.
Can we change the selector to just `a`, and ditch all of the pseudo-class selectors?

albachteng replied:
Thank you both @jrapala @daleinen7, I see what was happening now. I've removed the redundant rules and think we should be good to go

jrapala commented on scss/main.scss:
```diff
@@ -337,15 +374,15 @@ h1,h2,h3 {
   padding-right: .75rem;
   a {
-    color: $cfa-red !important;
+    color: $white !important;
```
Can remove this nested `a` selector? The `whitebutton` class is placed on an anchor tag so this style doesn't do anything.
